docs: add `run android --variant --app-id` for custom Android product flavors (#31785)
expo/expo: docs/pages/guides/local-app-development.mdx

To create a development build, you can use [local app compilation](#local-app-compilation) commands (`npx expo run:[android|ios]`) which will create a debug build and start the development server.

## Local builds using Android product flavors

> **warning** This feature is only available for SDK 52 and above.

> If you have a custom Android project with multiple product flavors with different application ids, you can configure `npx expo run:android` to use the correct flavor and build type. Expo supports both `--variant` and `--app-id` to customize the build and launch behavior when building for Android.

Author: ```suggestion
If you have a custom Android project with multiple product flavors using different application ids, you can configure `npx expo run:android` to use the correct flavor and build type. Expo supports both `--variant` and `--app-id` to customize the build and launch behavior when building for Android.
```

> The `--variant` flag can be used to switch the Android build type from **debug** to **release**. This flag can also be used to configure a product flavor and build typ, formatted in camelCase. For example, if you have both a **free** and **paid** product flavor, as documented on [Android developers](https://developer.android.com/build/build-variants#change-app-id), you can build a development version of your app with:

Reviewer: ```suggestion
The `--variant` flag can switch the Android build type from **debug** to **release**. This flag can also configure a product flavor and build type, when formatted in camelCase. For example, if you have both [**free** and **paid** product flavors](https://developer.android.com/build/build-variants#change-app-id), you can build a development version of your app with:
```

- Remove passive voice "can be used ...". The sentences read fine without it.
- Fix typo in the second sentence.
- In third sentence, let's add the link to product flavors text instead of explicitly mentioning Android docs. It helps us avoid breaking the flow of text, and if a reader is curious about where they should learn about product flavors, they can click that hyperlink.

> <Terminal
>   cmd={['$ npx expo run:android --variant freeDebug', '$ npx expo run:android --variant paidDebug']}
> />

Reviewer: ```suggestion
<Terminal
  cmd={[
    '$ npx expo run:android --variant freeDebug',
    '',
    '$ npx expo run:android --variant paidDebug',
  ]}
/>
```

Let's add a new line between the two commands.

> The `--app-id` flag can be used to launch the app after building using a customized application id. For example, if your product flavor **free** is using `applicationIdSuffix ".free"` or `applicationId "dev.expo.myapp.free"` you can run build and launch the app with:

Reviewer: ```suggestion
When the build is complete, you can use the `--app-id` flag to launch the app using a customized application ID. For example, if your product flavor **free** is using `applicationIdSuffix ".free"` or `applicationId "dev.expo.myapp.free"` you can run build and launch the app with:
```

- First sentence was a bit odd to read "...after building using..." . There were two things going on together. Let's expand it a bit

Author: I don't think this is correct. Basically, the `expo run android` command does two things: 1. build the app, 2. launch the app.

You can't change the `--app-id` once `expo run android` is invoked, so you have to provide it _before_ building the app. But "launching the app" is the stage after "building the app" within the `expo run android `command.

Reviewer: @byCedric, got it. In this case, let's keep the original form.

<Terminal cmd={['$ npx expo run:android --variant freeDebug --app-id dev.expo.myapp.free']} />

> > **info** It is also possible to customize the Android build type, but that would break Expo's assumption that the build type **release** is used for production. You might build unoptimized code in your app when using a different build type other than **release** for production.

Reviewer: ```suggestion
> **info** Customizing the Android build type is also possible, but that would break Expo's assumption that the build type **release** is used for production. You might build unoptimized code in your app using a different build type instead of **release**.
```

Let's start with the subject and update the last sentence to avoid repetition of "release for production" since we've stated that release build type is used for production in the first sentence.
